index.js

```diff
@@ -68,6 +68,17 @@ module.exports = function requireDir(dir, opts) {
             }
 
             if (FS.statSync(path).isDirectory()) {
+
+                // If we're not recursing through the files then we check to see
+                // if the folder is a valid node module
+
+                // Checks to see if there is an index file and returns the extension
+                var modulePath = getModulePath(path);
+
+                if (modulePath && !opts.recurse) {
+                    filesMinusDirs[file + Path.extname(modulePath)] = modulePath;
+                }
+
```

Comment:
(Assuming for now we keep your design that this should be the default behavior) Should this happen only if `opts.recurse` *isn't* specified?

Put another way: what should happen when `opts.recurse` *is* specified? Currently, we'd `require` the dir normally, and then that would get overwritten on recurse, is that right? So harmless, but inefficient?

Can we add a test case for what should happen when `opts.recurse` *is* specified and a (sub)directory is a module? Thanks!